Tests/SwiftLintFrameworkTests/ConfigurationTests+MultipleConfigs.swift

```diff
@@ -411,12 +411,159 @@ extension ConfigurationTests {
+            Configuration.emptyOnlyConfiguration(),
+            Configuration.enabledOnlyConfiguration(ruleIdentifier),
+            Configuration.allEnabledConfiguration()
+        ]
+
+        let configurations = [
+            Configuration(rulesMode: .default(disabled: [], optIn: [])),
+            Configuration(rulesMode: .default(disabled: [], optIn: [ruleIdentifier])),
+            Configuration(rulesMode: .default(disabled: [ruleIdentifier], optIn: [ruleIdentifier])),
+            Configuration(rulesMode: .default(disabled: [ruleIdentifier], optIn: []))
+        ]
+
+        for parentConfiguration in parentConfigurations {
+            testParentConfiguration(parentConfiguration, configurations: configurations, ruleType: ruleType)
+        }
+    }
+
+    private func validateConfiguredRuleIsEnabled(
+        parentConfiguration: Configuration?,
+        disabledRules: Set<String>,
+        optInRules: Set<String>,
+        ruleType: any Rule.Type
+    ) -> Issue? {
+        var enabledInParentRules: Set<String> = []
+        var disabledInParentRules: Set<String> = []
+        var allEnabledRules: Set<String> = []
+
+        if case .only(let onlyRules) = parentConfiguration?.rulesMode {
+            enabledInParentRules = onlyRules
+        } else if case .default(let parentDisabledRules, let parentOptInRules) = parentConfiguration?.rulesMode {
+            enabledInParentRules = parentOptInRules
+            disabledInParentRules = parentDisabledRules
+        }
+        allEnabledRules = enabledInParentRules
+            .subtracting(disabledInParentRules)
+            .union(optInRules)
+            .subtracting(disabledRules)
+
+        return Configuration.validateConfiguredRuleIsEnabled(
+            parentConfiguration: parentConfiguration,
+            enabledInParentRules: enabledInParentRules,
+            disabledInParentRules: disabledInParentRules,
+            disabledRules: disabledRules,
+            optInRules: optInRules,
+            allEnabledRules: allEnabledRules,
+            ruleType: ruleType
+        )
+    }
+
+    private func testParentConfiguration(
+        _ parentConfiguration: Configuration?,
+        configurations: [Configuration],
+        ruleType: any Rule.Type
+    ) {
+        for configuration in configurations {
+            if case .default(let disabledRules, let optInRules) = configuration.rulesMode {
+                let mergedConfiguration = parentConfiguration?.merged(withChild: configuration) ?? configuration
+                let isEnabled = mergedConfiguration.contains(rule: ruleType)
+                let issue = validateConfiguredRuleIsEnabled(
+                    parentConfiguration: parentConfiguration,
+                    disabledRules: disabledRules,
+                    optInRules: optInRules,
+                    ruleType: ruleType
+                )
+                XCTAssertEqual(isEnabled, issue == nil)
+                if let issue {
```

Comment:
Unwrapping shouldn't be needed if `issues` is exclusively used for comparisons in assertions. Leads to one less indentation ...

Reply:
So the code below does presume that `issue` is non-nil, but I've switched to a guard to reduce indentation